### ISSUE TITLE
How to use parallel_flat_hash_map

### ISSUE THREAD

**Issue by realwzc** [greg7mdp/parallel-hashmap#261]
Dear Author,

After reading your code, I have two questions I’d like to ask:

- When using parallel_flat_hash_map and passing a lock type through the template parameter, can the map still support operations like:

```
auto iter = map.find(key);  
if (iter != map.end()) {  
    map.erase(iter);  
}  
```
or
```
map[key] = value;  
```

- Does node_hash_map prevent crashes caused by rehashing during concurrent reads and writes of different keys? Our program can guarantee that different threads only read or write distinct keys.

Best regards,

**Comment by greg7mdp** [comment]
> * can the map still support operations like

No, these APIs cannot be made thread safe, but phmap has extended APIs which are thread safe (basically providing a lambda which is called under the lock protection). See this paragraph at the end of the README.

<img width="638" alt="image" src="https://github.com/user-attachments/assets/b775bc92-ac3f-46f9-a616-8d213aeca929">

**Comment by realwzc** [comment]
Thank you for your response, but if I want to delete this key, how should I do it? Is using `map.erase(key);` thread-safe?

**Comment by greg7mdp** [comment]
Yes, `erase(key)` is safe. If you want to check something before erasing the key, there is `erase_if`.
